src/ios/FirebaseMessagingPlugin.m

```diff
@@ -180,7 +180,12 @@ - (void)registerNotifications:(NSError *)error {
             pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:error.localizedDescription];
         } else {
             NSData* deviceToken = [FIRMessaging messaging].APNSToken;
-            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArrayBuffer:deviceToken];
+
+            if (deviceToken == nil) {
+                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Could not retrieve APNS token"];
```

Comment:
I don't think we should trigger error result here. `requestPersmission` optionally returns APNS token, so better to return just null is this case:
```js
pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:nil];
```

Reply:
Implemented as requested